### PR TITLE
Fixes #1 - Allows use of re and fullMatch outside of main module

### LINKED ISSUE #1: re and fullMatch do not work outside main module on Nim devel
Due to step and toPattern not being public.

### PR DESCRIPTION
**Pull request by honewatson**

### CODE REVIEW

File: src/regex.nim [hunk 1]
```diff
@@ -1566,7 +1566,7 @@ iterator peek[T: seq[Rune] | string](s: T): (int, Rune, Rune) {.inline.} =
   yield (j, prev, invalidRune)
 
 # todo: make it iterative
-proc step(
+proc step*(
```
Comment (nitely) [inline src/regex.nim]: are you sure this must be public?

**Comment by nitely** [comment]
Hey @honewatson, glad someone is already trying this 😄 

There are a ton of test within the module. I was going to split them into tests modules (see the comments named `tsome`) and avoid using the `unittest` module in case this ever gets into Nim's stdlib. Of course I'm still working on it, @dom96 made me release it before it was stable 🤣 , so now I have to prioritize things (API stability first).

File: src/regex.nim (hunk 1 above)
Comment (nitely) [reply src/regex.nim]: I just tried this. The issue is in the `template stepFrom()` below. It seems the method syntax has some limitations and can't be resolved from `result.step(...)` to `step(result, ...)`, but changing that fixes the issue.

Comment (nitely) [reply src/regex.nim]: ok, just fixed it this way ^^

Comment (nitely) [reply src/regex.nim]: I reported the issue https://github.com/nim-lang/Nim/issues/7085

**Comment by nitely** [comment]
I've fixed most issues here (without exporting non-public APIs). I'd like to keep the test, but not sure if using `unittest` is a good idea coz of what I commented above. If you can reimplement it with good old `doAssert` somwhere within the regex module, I'll accept the PR, but if you lack time for it, feel free to close it and I'll do it at some point. Thanks!

**Comment by honewatson** [comment]
Not sure how koch testing works but I do think that tests should be independent of the source.

**Comment by honewatson** [comment]
Anyway terrific work and well done.  Its nice to see it working with the complex email regex out of the box.

**Comment by nitely** [comment]
> Not sure how koch testing works but I do think that tests should be independent of the source.

I agree. The tests must be moved to a test folder, like you did here. In fact there are comments the future names of the files (i.e: `tescaped_sequences.nim`). What I'd prefer is not relying on the `unittest` module, just in case there is a chance this will make it into stdlib. But I guess we could use `unittest` and then make changes one day if necessary.

> Anyway terrific work and well done. Its nice to see it working with the complex email regex out of the box.

Thanks!! 😄

**Comment by nitely** [comment]
I'm closing this. I've created a tests folder and moved almost all tests there. I'd like to keep the email test from this PR. So if you feel like it, please open a new PR with the test case, but there's no rush to do it. Thanks!

**Pull request closed without merging**